core/src/main/scala/org/apache/spark/deploy/master/ApplicationInfo.scala

```diff
@@ -32,7 +32,8 @@ private[spark] class ApplicationInfo(
     val desc: ApplicationDescription,
     val submitDate: Date,
     val driver: RpcEndpointRef,
-    defaultCores: Int)
+    defaultCores: Int,
+    var appUiUrl: String)
```

Comment:
Yes, I moved it to that class because it represent a mutable state of application in the Master - and if the appUiUrl changes, it is due to application state change. This class has many more mutable fields so it doesn't make it worse. The main purpose of this ticket was to move out mutable fields from `ApplicationDescription` which is used to transfer data from one process to another and by definition it should be immutable.